packages/studioCMS/src/integrations/studioCMSDashboard/routes/dashboard/components/webVitalsUtils.ts

```diff
@@ -0,0 +1,252 @@
+    const lcpData: number[] = [];
+    if (webVitalData) {
+        for (const item of webVitalData) {
+            if (item.name === "LCP") {
+                lcpData.push(item.value);
+            }
+        }
+    }
+    return calculateLcpAverage(lcpData);
+};
+
+export function calculateLcpScoreText(lcp: number): string {
+    if (msToSeconds(lcp) <= 2) {
+        return "Excellent"; // Excellent
+    }if (msToSeconds(lcp) <= 4) {
+        return "Good"; // Good to Excellent
+    }if (msToSeconds(lcp) <= 6) {
+        return "Fair"; // Fair to Good
+    }
+    return "Poor"; // Poor to Fair
+}
+
+export function calculateLcpScorePercent(lcp: number): number {
+    if (msToSeconds(lcp) <= 2) {
+        return 100; // Excellent
+    }if (msToSeconds(lcp) <= 4) {
+        return Math.round(100 - ((lcp - 2) / (4 - 2)) * 50); // Good to Excellent
+    }if (msToSeconds(lcp) <= 6) {
+        return Math.round(50 - ((lcp - 4) / (6 - 4)) * 50); // Fair to Good
+    }
+        return Math.round(0 - ((lcp - 6) / (10 - 6)) * 50); // Poor to Fair
+}
+
+export const progressBarLcpColor = (lcpData: number) => {
+    if (msToSeconds(lcpData) <= 2.5) {
+        return "green";
+    } 
+    if (msToSeconds(lcpData) > 2.5 && lcpData <= 4) {
+        return "yellow";
+    } 
+    return "red";
+};
+
+export const progressBarLcpTrackColor = (lcpData: number) => {
+    if (msToSeconds(lcpData) <= 2.5) {
+        return "yellow";
+    } 
+    if (msToSeconds(lcpData) > 2.5) {
+        return "red";
+    } 
+    return "red";
+};
+
+export const lcpTextColor = (lcpData: number) => {
+    if (msToSeconds(lcpData) <= 2.5) {
+        return "green";
+    } 
+    if (msToSeconds(lcpData) > 2.5 && lcpData <= 4) {
+        return "yellow";
+    } 
+    return "red";
+};
+
+// INP Calculations
+export function calculateInpAverage(inpValues: number[]): number {
+
+    const sum = inpValues.reduce((acc, curr) => acc + curr, 0);
+    const average = sum / inpValues.length;
+    
+    // Round average to two decimal places
+    return Math.round(average * 100) / 100;
+}
+
+export function inpDataAverage(webVitalData: WebVitalsResponseItem[]): number {
+    const inpData: number[] = [];
+    if (webVitalData) {
+        for (const item of webVitalData) {
+            if (item.name === "INP") {
+                inpData.push(item.value);
+            }
+        }
+    }
+    return Math.floor(calculateInpAverage(inpData));
+}
+
+export function calculateInpScoreText(inp: number): string {
+    if (inp <= 50) {
+        return "Excellent"; // Excellent
+    }if (inp <= 100) {
+        return "Good"; // Good to Excellent
+    }if (inp <= 200) {
+        return "Fair"; // Fair to Good
+    }
+    return "Poor"; // Poor to Fair
+}
+
+export function calculateInpScorePercent(inp: number): number {
+    if (inp <= 50) {
+        return 100; // Excellent
+    }if (inp <= 100) {
+        return Math.round(100 - ((inp - 50) / (100 - 50)) * 50); // Good to Excellent
```

Comment:
And these

Reply:
https://web.dev/articles/inp

Reply:
[2e5f04a](https://github.com/astrolicious/studiocms/pull/125/commits/2e5f04a92da69eb44ba45385409c4d99014b2878) comments